XenModel/Actions/Updates/DownloadAndUpdateClientAction.cs

```diff
@@ -190,14 +191,14 @@ protected override void Run()
 
         public void InstallMsi()
         {
-            ValidateMsi();
 
             if (!File.Exists(outputPathAndFileName))
                 throw new Exception(Messages.DOWNLOAD_CLIENT_INSTALLER_MSI_NOT_FOUND);
 
             // Install the msi            
             try
             {
+                ValidateMsi();
                 // Start the install process, it will handle closing of application.
                 Process.Start(outputPathAndFileName);
                 log.DebugFormat("Update {0} found and install started", updateName);

```

Comment:
Nit, not related to this PR. To avoid refactoring errors, I'd do `OnInstall?.Invoke()`

Reply:
Is that within the ValidateMsi function?